src/validations/rules/sap.sch

```diff
@@ -171,6 +194,37 @@
 
     </sch:pattern>
 
+    <sch:pattern
+        id="assessment-subject">
+        <sch:rule
+            context="oscal:assessment-subject[@type = 'location']">
+            <sch:assert
+                diagnostics="location-not-include-all-element-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.3"
+                fedramp:specific="true"
+                id="location-not-include-all-element"
+                role="error"
+                test="not(exists(oscal:include-all))">The FedRAMP SAP document must explicitly cite locations.</sch:assert>
+        </sch:rule>
+        <sch:rule
+            context="oscal:include-subject[@type='location']">
+            <sch:let
+                name="ssp-locations"
+                value="$ssp-doc/oscal:system-security-plan/oscal:metadata//oscal:location/@uuid ! xs:string(.)" />
+            <sch:let
+                name="sap-locations"
+                value="/oscal:assessment-plan/oscal:metadata//oscal:location/@uuid ! xs:string(.)" />
+            <sch:assert
+                diagnostics="location-uuid-matches-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.3, §4.3.1"
+                fedramp:specific="true"
+                id="location-uuid-matches"
+                role="error"
+                test="@subject-uuid[. = $ssp-locations] or @subject-uuid[. = $sap-locations]"
```

Comment:
I _think_ this simplifies to `test="@subject-uuid = $ssp-locations  or @subject-uuid = $sap-locations"`.

Reply:
That appears to work the same as the previous code.  I've simplified as suggested.